fix: Extend allowed characters to DomValidator

**Description**

Take the following, valid, [`Alpine.js`](https://alpinejs.dev/start-here#listening-for-a-click-outside) use case:

```html
<div ... x-on:click.outside="open = false">Contents...</div>
```

This will throw a server error:

```
[SERVER] [ERROR] Invalid argument(s): "x-on:click.outside" is not a valid attribute name.
[SERVER] [ERROR] package:jaspr/src/server/markup_render_object.dart 165:7   DomValidator.validateAttributeName
```

**Doctor Output**

Using the latest `main` branch revisions of all `jaspr` apps.

**Discussion**

```
packages/jaspr/lib/src/server/markup_render_object.dart
```

The following modification should resolve the issue:

```dart
  static final _attributeRegExp =
      RegExp(r'^[a-z](?:[a-zA-Z0-9\-_:.]*[a-z0-9]+)?$');
```

This modification allows for `.` to appear anywhere in the string along with the other special characters.

With this change, the following string is now valid: `x-on:click.outside`

--- a/packages/jaspr/CHANGELOG.md
+++ b/packages/jaspr/CHANGELOG.md
@@ -1,3 +1,7 @@
+## Unreleased patch
+
+- Fixed `DomValidator` to allow attributes with `.`.
+
 ## 0.11.1
 
 - Fixed bug with base paths.

--- a/packages/jaspr/lib/src/server/markup_render_object.dart
+++ b/packages/jaspr/lib/src/server/markup_render_object.dart
@@ -126,7 +126,7 @@ class MarkupRenderObject extends RenderObject {
 
 /// DOM validator with sane defaults.
 class DomValidator {
-  static final _attributeRegExp = RegExp(r'^[a-z](?:[a-zA-Z0-9\-_:]*[a-z0-9]+)?$');
+  static final _attributeRegExp = RegExp(r'^[a-z](?:[a-zA-Z0-9\-_:.]*[a-z0-9]+)?$');
   static final _elementRegExp = _attributeRegExp;
   static const _selfClosing = <String>{
     'area',

